### ISSUE TITLE
Array shapes support

### ISSUE THREAD

**Issue by Kiina** [zircote/swagger-php#1676]
I've been trying to get my custom processor to get a good swagger.json for arrays. I've tried to use the PhpDocParser but noticed it gets a bit complex after a while. Then I looked at how doctypes are handled by and saw the [extractVarTypeAndDescription](https://github.com/zircote/swagger-php/blob/7df10e8ec47db07c031db317a25bef962b4e5de1/src/Processors/Concerns/DocblockTrait.php#L177) method. It uses some regex that won't really work with [array shapes](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#array-shapes).

Let's take a simple one:

`/* @var array{id: int, name: string} */`

Calling the extract on it returns:

```php
array(2) {
  ["type"]=>
  string(9) "array{id:"
  ["description"]=>
  string(18) "int, name: string}"
}
```

Which results in:

```yml
properties:
  name:
    description: 'int, name: string}'
```

Which isn't the desired result.

**Comment by DerManoMann** [comment]
Fair point. I think I mayredo that for v5 still, but I am not inclined to write more ad-hoc docblock parsing code.
PRs are always welcome, of course.

**Comment by Kiina** [comment]
Current behavior could be preserved pretty easily:

```php
function extractVarTypeAndDescription(?string $docblock): array
{
    $config = new ParserConfig([]);
    $lexer = new Lexer($config);
    $constExprParser =  new ConstExprParser($config);
    $parser = new PhpDocParser($config, new TypeParser($config, $constExprParser), $constExprParser);

    $tokens = new TokenIterator($lexer->tokenize($docblock));
    $phpDocNode = $parser->parse($tokens);
    $varTags = $phpDocNode->getVarTagValues();

    // Not sure if a var tag can have more than 1?
    foreach ($varTags as $varTag) {
        if($varTag->type instanceof IdentifierTypeNode){
            return ['type' => $varTag->type->name, 'description' => $varTag->description?:null];
        }
    }
}
```

But there probably need to be changes in some more places to handle cases like NullableTypeNode and UnionTypeNode. Since the mapNativeType is also a bit rudimentary

**Comment by DerManoMann** [comment]
Yep, gets tricky pretty quick. Good for adding support for multiple types, though.

**Comment by DerManoMann** [comment]
https://github.com/symfony/symfony/pull/59676 would allow to use (a backported) the Symfony `TypeInfo` component for all docblock type handling...
